trio/_subprocess.py

```diff
@@ -439,17 +470,18 @@ async def _posix_deliver_cancel(p):
         )
 
 
-async def run_process(
-    command,
+# Use a private name, so we can declare platform-specific stubs below.
+async def _run_process(
+    command: StrOrBytesPath | Sequence[StrOrBytesPath],
     *,
-    stdin=b"",
-    capture_stdout=False,
-    capture_stderr=False,
-    check=True,
-    deliver_cancel=None,
-    task_status=trio.TASK_STATUS_IGNORED,
-    **options,
-):
+    stdin: bytes | bytearray | memoryview | int | HasFileno | None = b"",
+    capture_stdout: bool = False,
+    capture_stderr: bool = False,
+    check: bool = True,
+    deliver_cancel: Callable[[Process], Awaitable[object]] | None = None,
+    task_status: TaskStatus = trio.TASK_STATUS_IGNORED,  # type: ignore[assignment]  # TODO
```

Comment:
bunch of bare `# TODO`s without any explanation - but I assume the majority of them are just waiting for #2733

Reply:
Yeah these are just reminders to make `TaskStatus` generic.

Reply:
I've merged #2733 so you can go ahead with it now